### ISSUE TITLE
Package install fails because the install directory is deleted

### ISSUE THREAD

**Issue by lgeiger** [atom/atom#13256]
### Prerequisites

* [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode)?
* [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
* [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
* [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
* [ ] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
I reported this to https://github.com/atom/apm/issues/648 as well.
* [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

### Description

We are running into big issues in Hydrogen because the install directory get's deleted before the package is fully installed. This doesn't happen on every system and it shouldn't be a package specific problem. We'd really appreciate your help.

Here is the full log: [hydrogen-install.log.zip](https://github.com/nteract/hydrogen/files/596086/hydrogen-install.log.zip)

### Steps to Reproduce

1. `apm install hydrogen`

**Expected behavior:**

Hydrogen installs successfully.

**Actual behavior:**

The installation aborts half way, because the temporary directory is deleted before the installation finished.

### Versions

```
$ apm --version
apm  1.12.9
npm  3.10.5
node 4.4.5
python 2.7.12
git 2.10.2
```
This isn't a operating specific problem. We have reports about this on Windows, Linux and Mac OS X.

**Comment by andyrichardson** [comment]
I wasn't able to replicate this on my system - 

OS: Linux

```
$ apm --version
apm  1.12.9
npm  3.10.5
node 4.4.5
python 2.7.12
git 2.9.3
```

**Comment by lgeiger** [comment]
@andyrichardson Thanks for testing.

We resolved the issue in Hydrogen `v1.2.0` by shipping prebuilt binaries of [zeromq](https://github.com/zeromq/zeromq.js) for the usage in Electron instead of building the native module from source during the install.
This heavily reduces the install time for users and therefor works around this issue.
